dubbo-common/src/main/java/org/apache/dubbo/common/compiler/support/JavassistClassInfo.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.common.compiler.support;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javassist.CannotCompileException;
+import javassist.ClassPool;
+import javassist.CtClass;
+import javassist.CtField;
+import javassist.CtNewConstructor;
+import javassist.CtNewMethod;
+import javassist.NotFoundException;
+
+/**
+ * Javassist class info contains all the information used for JavassistCompiler, including:
+ * <p>
+ * class name, imported packages, super class name, implemented interfaces, constructors, fields, methods.
+ */
+public class JavassistClassInfo {

```

Comment:
As once this class gets initialized then it is not getting modified, I would recommend to use builder for this and then have only getter or make it immutable. 

have addInterface, addConstructor keep it the door open for accidental bug. What do you say?

Reply:
could you give more detailed information what shall i do, could be some pseudo code or java code.
thank you

Reply:
it was saying having setter for collections e.g. setImports(List<String> imports), setInterfaces(List<String> ifaces) .
but looking at again it seems you are doing more than just setter. In this case I think it is ok to have this.

Reply:
thanks